### PR TITLE
chore: refactor info package

### PR DESCRIPTION
**Pull request by rishabh3112**
**What kind of change does this PR introduce?**
refactor/feature.
- Display all info and remove unnecessary flags

**Did you add tests for your changes?**
WIP

**If relevant, did you update the documentation?**
WIP

**Summary**
Would be adding info package as direct dependency as per roadmap

**Does this PR introduce a breaking change?**
No

**Other information**
No

### CODE REVIEW

File: test/info/info-help.test.js [hunk 1]
```diff
@@ -28,13 +28,6 @@ describe('should print help for info command', () => {
...
-    it('should output help even if other flags are supplied with help', () => {
-        const { stdout, stderr } = runInfo(['info', '--help', '--system']);
-        expect(stdout).toContain(usageText);
-        expect(stdout).toContain(descriptionText);
-        expect(stdout).not.toContain('System:');
-        expect(stderr).toHaveLength(0);
-    });
 
     it('should respect the --color=false flag', () => {
         const { stdout, stderr } = runInfo(['info', 'help', '--color=false']);
```
Comment (evilebottnawi) [inline test/info/info-help.test.js]: Please improve tests, we should check what it is included `npm`/`yarn`/`webpack`/etc strings

File: packages/info/src/index.ts [hunk 2]
```diff
@@ -13,13 +13,6 @@ interface Information {
...
-interface ArgvI {
-    _?: string[];
-    bin?: boolean;
-    binaries?: boolean;
-    config?: string;
-}
-
 const DEFAULT_DETAILS: Information = {
     Binaries: ['Node', 'Yarn', 'npm'],
     Browsers: ['Chrome', 'Firefox', 'Safari'],
```
Comment (evilebottnawi) [inline packages/info/src/index.ts]: Need add all possible browsers

File: packages/info/src/options.ts [hunk 3]
```diff
@@ -1,42 +1,7 @@
...
-		demandOption: false,
-		describe: "To get the output as markdown",
-		type: "boolean"
-	})
-	.group(["output-json", "output-markdown"], `Output format`).argv;
+export default [
+    {
+        name: 'output',
+        type: String,
+        description: 'To get the output in specified format ( accept json or markdown )',
```
Comment (evilebottnawi) [inline packages/info/src/options.ts]: Avoid spaces after `(` and before `)`

File: test/info/info-output.test.js [hunk 4]
```diff
@@ -18,7 +18,7 @@ describe('basic info usage', () => {
     });
 
     it('gets info as json', async () => {
-        const { stdout, stderr } = await runInfo(['--output-json']);
+        const { stdout, stderr } = await runInfo(['--output="json"']);
         expect(stdout).toContain('"System":');
         expect(stderr).toHaveLength(0);
```
Comment (evilebottnawi) [inline test/info/info-output.test.js]: Please add test what output is valid JSON - `JSON.parse` should not throw an error and contains basic fields

File: test/info/info-help.test.js (hunk 1 above)
Comment (evilebottnawi) [reply test/info/info-help.test.js]: Not only for that test

File: packages/info/src/index.ts [hunk 5]
```diff
@@ -28,55 +21,30 @@ const DEFAULT_DETAILS: Information = {
...
-            return { npmGlobalPackages: DEFAULT_DETAILS.npmGlobalPackages };
-        case 'npmPackages':
-            return { npmPackages: DEFAULT_DETAILS.npmPackages };
-    }
-}
-
-export default async function info(customArgv: object): Promise<string[]> {
-    let detailsObj = {};
+export default async function info(args: object): Promise<string[]> {
+    console.log(args);
```
Comment (anshumanv) [inline packages/info/src/index.ts]: rm logs

File: test/info/info-help.test.js (hunk 1 above)
Comment (rishabh3112) [reply test/info/info-help.test.js]: This tests is for help flag used with `info`. But will add these in the tests for `info` output.

File: test/info/info-output.test.js (hunk 4 above)
Comment (rishabh3112) [reply test/info/info-output.test.js]: JSON parse test exist already ( Just below were you have made comment. )
Would add tests for basic fields.

File: packages/info/src/index.ts [hunk 6]
```diff
@@ -13,70 +12,50 @@ interface Information {
...
-    bin?: boolean;
-    binaries?: boolean;
-    config?: string;
-}
-
 const DEFAULT_DETAILS: Information = {
     Binaries: ['Node', 'Yarn', 'npm'],
-    Browsers: ['Chrome', 'Firefox', 'Safari'],
+    Browsers: [
+        'Brave Browser',
```
Comment (evilebottnawi) [inline packages/info/src/index.ts]: `Brave Browser`? What is it? :smile:

Comment (rishabh3112) [reply packages/info/src/index.ts]: https://brave.com/
A browser form a company of Brendan Eich ✨

**Comment by rishabh3112** [comment]
Snapshots for testing is a bad idea IMO. On different testing environments (windows, linux, macOS, ...) output would be different on each one of them.

Here is output:
image

image

image

I use WSL and don't have any browser installed over there, so no browsers :)

**Comment by evilebottnawi** [comment]
> Snapshots for testing is a bad idea IMO. On different testing environments (windows, linux, macOS, ...) output would be different on each one of them.

No, it is good idea, we should remove platform specific output from snapshot (like do it webpack)

**Comment by rishabh3112** [comment]
@evilebottnawi **System** output is very important for info command, and it is system specific IMO.

**Comment by evilebottnawi** [comment]
Now I have to spar every time what the output looks like, it’s extremely inconvenient and I can skip some regressions, run it locally every time and look very time-consuming, so we need to simplify this

**Comment by evilebottnawi** [comment]
@rishabh3112 I will improve tests in future and you see how it is better, let's avoid it here, just provide example current default output

**Comment by rishabh3112** [comment]
Here it is https://github.com/webpack/webpack-cli/pull/1382#issuecomment-606588072

**Comment by rishabh3112** [comment]
@evilebottnawi also should I add the `info` as direct dependency in this PR itself, or should we release it `info` first to npm with current changes?

**Comment by evilebottnawi** [comment]
Let's rename `npmPackages:` to `Packages:` and `globalNpmPackages` to `Global packages:`

> @evilebottnawi also should I add the info as direct dependency in this PR itself, or should we release it info first to npm with current changes?

In the next PR

**Comment by rishabh3112** [comment]
Replaced output:
image

File: packages/info/src/index.ts (hunk 6 above)
Comment (evilebottnawi) [reply packages/info/src/index.ts]: Do you know who use it? :smile:

Comment (rishabh3112) [reply packages/info/src/index.ts]: I use it (+edge chromium).
https://brave.com/brave-passes-10m-mau/

Comment (rishabh3112) [reply packages/info/src/index.ts]: Nevertheless, not a blocker for this PR anyways.

**Comment by evilebottnawi** [comment]
Thanks, feel free to integrate env package into CLI

**Pull request merged**